packages/webpack-cli/lib/utils/cli-executer.js

```diff
@@ -0,0 +1,53 @@
+const { MultiSelect, Input} = require('enquirer');
+const runner = require('../runner');
+const logger = require('./logger');
+const allArgs = require('./cli-flags');
+
+logger.info(`Initiating webpack-cli executer \n`);
+
+const cliArgs = [];
+const nodeArgs = [];
+async function prompter() {
+    const typePrompt = new MultiSelect({
+        name: 'type',
+        message: 'Which flags do you want to use ?',
+        choices: allArgs.core.reduce((prev, curr) => {
+            return [...prev, `--${curr.name}`];
+        }, [])
+    });
+
+    const selections = await typePrompt.run();
+    const questions = selections.map(selection => {
```

Comment:
Change approach in this function:
1. create a local variable where you store args and `Input`s
2. iterate `selections` using a `.forEach` (you need a local variable to store `Input` instances) or `.reduce` (here you return an array of `Input`)
3. If you find a `Boolean` you don't return anything and you its value inside the local variable for storing the args. You don't create that `run: () =>` which I don't like. It's basically smell code.
4. Iterate over the `Input` array using a `async .map` where you return the cli arg value, together with a `Promise.all`

```js
const args = await Promise.all(questions.map(async question => {
   const value = await question.run();
  // do something here
   return valueAsArg;
}));

return [
  ...boolArgs,
  ...args
]
```

Reply:
Can't use `Promise.all` as it will force all the promises to execute in parallel ( which in turn will be blocked by await ), that is why, I used reduce to create chain for synchrunous execution.
I have taken care of all other issues you have mentioned.

Reply:
It makes. So at this point consider the following approaches:
1. A queue
2. An asynchronous loop (you will have to use a map/set)

Reply:
Just curious, what you meant by 2nd approach?